packages/grafana-ui/src/components/FileDropzone/FileDropzone.tsx

```diff
@@ -0,0 +1,224 @@
+        setFiles((oldFiles) => [...oldFiles, ...customFiles]);
+      }
+
+      if (options?.onDrop) {
+        options.onDrop(acceptedFiles, rejectedFiles, event);
+      } else {
+        for (const customFile of customFiles) {
+          const reader = new FileReader();
+
+          const read = () => {
+            if (readAs) {
+              reader[readAs](customFile.file);
+            } else {
+              reader.readAsText(customFile.file);
+            }
+          };
+
+          // Set abort FileReader
+          setFileProperty(customFile, (fileToModify) => {
+            fileToModify.abortUpload = () => {
+              reader.abort();
+            };
+            fileToModify.retryUpload = () => {
+              setFileProperty(customFile, (fileToModify) => {
+                fileToModify.error = null;
+                fileToModify.progress = undefined;
+              });
+              read();
+            };
+          });
+
+          reader.onabort = () => {
+            setFileProperty(customFile, (fileToModify) => {
+              fileToModify.error = new DOMException('Aborted');
+            });
+          };
+
+          reader.onprogress = (event) => {
+            setFileProperty(customFile, (fileToModify) => {
+              fileToModify.progress = event.loaded;
+            });
+          };
+
+          reader.onload = () => {
+            onLoad?.(reader.result);
+          };
+
+          reader.onerror = () => {
+            setFileProperty(customFile, (fileToModify) => {
+              fileToModify.error = reader.error;
+            });
+          };
+
+          read();
+        }
+      }
+    },
+    [onLoad, options, readAs, setFileProperty]
+  );
+
+  const removeFile = (file: DropzoneFile) => {
+    const newFiles = files.filter((f) => file.id !== f.id);
+    setFiles(newFiles);
+  };
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({ ...options, onDrop });
+  const theme = useTheme2();
+  const styles = getStyles(theme, isDragActive);
+  const fileList = files.map((file) => <FileListItem key={file.id} file={file} removeFile={removeFile} />);
+
+  return (
+    <div className={styles.container}>
+      <div {...getRootProps({ className: styles.dropzone })}>
+        <input {...getInputProps()} />
+        {children ?? <FileDropzoneDefaultChildren primaryText={getPrimaryText(files, options)} />}
+      </div>
+      {options?.accept ? (
+        <small className={cx(styles.small, styles.acceptMargin)}>{getAcceptedFileTypeText(options)}</small>
+      ) : null}
+      {fileList}
+    </div>
+  );
+}
+
+export function FileDropzoneDefaultChildren({ primaryText = 'Upload file' }) {
+  const theme = useTheme2();
+  const styles = getStyles(theme);
+
+  return (
+    <div className={styles.iconWrapper}>
+      <Icon name="upload" size="xxl" />
+      <h3>{primaryText}</h3>
+      <small className={styles.small}>
+        Drag and drop here or <span className={styles.link}>Browse</span>
```

Comment:
since you can interact with the whole area, i don't think we should pretend this is actually a link. what about giving `cursor: pointer;` to the dropzone class and changing the text to `Drag and drop here or click to browse`?

Reply:
Sounds good to me!